github/repos_environments.go

```diff
@@ -168,6 +168,13 @@ type CreateUpdateEnvironment struct {
 	DeploymentBranchPolicy *BranchPolicy   `json:"deployment_branch_policy"`
 }
 
+// CreateUpdateEnvironmentWithoutEnterprise represents the fields accepted for Pro/Teams private repos.
+// Ref: https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment
+// See https://github.com/google/go-github/issues/2602 for more information.
+type CreateUpdateEnvironmentWithoutEnterprise struct {
```

Comment:
This should not be exported as it is only used internally.

```suggestion
type createUpdateEnvironmentWithoutEnterprise struct {
```